src/benchmarks/gc/GC.Infrastructure/GC.Analysis.API/DynamicEvents/DynamicEvents.cs

```diff
@@ -43,69 +43,85 @@ public sealed class DynamicEventSchema
 
         public int MaxOccurrence { get; init; } = 1;
 
-        public static void Set(List<DynamicEventSchema> dynamicEventSchemas, bool allowPartialSchema = true)
+        static DynamicEventSchema()
         {
-            DynamicEventSchemas.Clear();
-            DynamicEventSchema.allowPartialSchema = allowPartialSchema;
-            foreach (DynamicEventSchema dynamicEventSchema in dynamicEventSchemas)
+            // One time add.
+            if (DynamicEventSchemas.Count == 0)
+            {
+                // Add the adaptation schemas.
+                Add(GCDynamicEvents.SizeAdaptationSampleSchema);
+                Add(GCDynamicEvents.SizeAdaptationTuningSchema);
+            }
+        }
+
+        public static void Add(DynamicEventSchema dynamicEventSchema, bool allowPartialSchema = true)
+        {
+            if (DynamicEventSchemas.ContainsKey(dynamicEventSchema.DynamicEventName))
+            {
+                throw new Exception($"Provided schema has a duplicated event named {dynamicEventSchema.DynamicEventName}");
+            }
+            CompiledSchema schema = new CompiledSchema();
+            if (dynamicEventSchema.MinOccurrence < 0)
+            {
+                throw new Exception($"Provided event named {dynamicEventSchema.DynamicEventName} has a negative MinOccurrence");
+            }
+            if (dynamicEventSchema.MaxOccurrence < dynamicEventSchema.MinOccurrence)
             {
-                if (DynamicEventSchemas.ContainsKey(dynamicEventSchema.DynamicEventName))
+                throw new Exception($"Provided event named {dynamicEventSchema.DynamicEventName} has a MaxOccurrence smaller than MinOccurrence");
+            }
+            schema.MinOccurrence = dynamicEventSchema.MinOccurrence;
+            schema.MaxOccurrence = dynamicEventSchema.MaxOccurrence;
+            int offset = 0;
+            foreach (KeyValuePair<string, Type> field in dynamicEventSchema.Fields)
```

Comment:
just verifying - these fields aren't aligned, correct?

Reply:
Yes, these aren't aligned - see: GCDynamicEvents.cs (the file I added) as reference.